api4/user.go

```diff
@@ -1129,6 +1140,93 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(ruser.ToJson()))
 }
 
+func patchUserCheckLoginProviderAttributes(c *Context, ouser *model.User, patch *model.UserPatch) {
+	const (
+		firstNameField = "first_name"
+		lastNameField  = "last_name"
+		nicknameField  = "nickname"
+		positionField  = "position"
+		emailField     = "email"
+		usernameField  = "username"
+	)
+	// If LDAP or SAML is used, and any of the following attributes have been specified
+	// in the settings, then their corresponding fields may not be changed
+	LS := &c.App.Config().LdapSettings
+	SS := &c.App.Config().SamlSettings
+	for _, provider := range []struct {
+		AuthService        string
+		FirstNameAttribute *string
+		LastNameAttribute  *string
+		NicknameAttribute  *string
+		PositionAttribute  *string
+		EmailAttribute     *string
+	}{
+		{model.USER_AUTH_SERVICE_LDAP, LS.FirstNameAttribute, LS.LastNameAttribute, LS.NicknameAttribute,
+			LS.PositionAttribute, LS.EmailAttribute},
+		{model.USER_AUTH_SERVICE_SAML, SS.FirstNameAttribute, SS.LastNameAttribute, SS.NicknameAttribute,
+			SS.PositionAttribute, SS.EmailAttribute},
```

Comment:
Are there any differences between the logic applied for LDAP vs SAML? If there are no differences, what are your thoughts on whether this can be done in a single loop or is it best to keep two loops?

Reply:
There are no differences, but one loop would be simpler.